tests/parametric/test_128_bit_traceids.py

```diff
@@ -427,30 +429,31 @@ def test_w3c_128_bit_propagation_tid_consistent(self, test_agent, test_library):
 
     @missing_feature(context.library == "ruby", reason="not implemented")
     @missing_feature(context.library == "nodejs", reason="not implemented")
-    @missing_feature(context.library == "dotnet", reason="not implemented")
     @missing_feature(context.library == "java", reason="not implemented")
     @pytest.mark.parametrize(
         "library_env",
         [{"DD_TRACE_PROPAGATION_STYLE": "tracecontext", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": "true",}],
     )
-    def test_w3c_128_bit_propagation_tid_in_parent(self, test_agent, test_library):
+    def test_w3c_128_bit_propagation_tid_in_chunk_root(self, test_agent, test_library):
         """Ensure that only root span contains the tid.
         """
         with test_library:
             with test_library.start_span(name="parent", service="service", resource="resource") as parent:
-                with test_library.start_span(name="child", service="service", parent_id=parent.span_id):
+                with test_library.start_span(name="child", service="service", parent_id=parent.span_id) as child:
                     pass
 
-        traces = test_agent.wait_for_num_traces(1, clear=True)
-        parent = find_span_in_traces(traces, Span(name="parent", service="service"))
-        child = find_span_in_traces(traces, Span(name="child", service="service"))
+        traces = test_agent.wait_for_num_traces(1, clear=True, sort_by_start=False)
+        trace = find_trace(traces, parent.trace_id)
+        assert len(trace) == 2
+        chunk_root = find_chunk_root_span(trace)
+        non_chunk_root = trace[1]
```

Comment:
There's an inconsistency here with trying to find the root span with `find_chunk_root_span()`, but using the span position to find the child span. The ordering of parent and child spans is not guaranteed. 

The `_dd.p.tid` span tag should always be added to the first span in the payload. This first span is not guaranteed to be the root span, so we should rely on span position only to test this tag.

I think this test do something like (consider this pseudo-code)
```python
# assert that "_dd.p.tid" is added only to first span in payload
assert trace[0]["meta"].get("_dd.p.tid") is not None
assert trace[1]["meta"].get("_dd.p.tid") is None
```

----

The `_dd.propagation_error` tag is another story. It is an optional tag that may be added to the root span while decoding headers. I don't think we should test it in this test. We already have a test above (`test_datadog_128_bit_propagation_tid_malformed_optional_tag`) that checks for its presence when a header is malformed. We could add another test to test for its absence when a header is valid, and this test would assert that the tag is missing on every span in the trace.

Reply:
Ahhh makes sense. I thought the goal of this test was to validate the first span in the trace payload, not necessarily the chunk root span. I just updated all instances were traces were accessed directly. All tests that validate the first span in a trace now use `find_chunk_root`.

Reply:
I think we're getting a few concepts mixed up here. Sorry if I wasn't clear before 😓 
 
> I thought the goal of this test was to validate the first span in the trace payload, not necessarily the chunk root span

Yes, that was correct: `_dd.p.tid` should be added to the first span in each trace chunk, which may or may not be the trace's root span. To check for this tag, we should rely on the order in which the tracer sent the spans in each chunk, not the relationship between spans (root, parent ,child).

>... not necessarily the chunk root span.
> All tests that validate the first span in a trace now use `find_chunk_root`.

What do you mean by "chunk root"? That's not a thing. The spans found in a trace chunk have no implied relationship (root, parent, child, etc) except that they all belong to the same trace.

Reply:
Talked on zoom and cleared up my confusion. Thanks! (resolving this thread)